pkg/operator/hostendpointscontroller2/host_endpoints_controller.go

```diff
@@ -117,6 +122,21 @@ func (c *HostEndpoints2Controller) syncHostEndpoints2(ctx context.Context, recor
 		})
 	}
 
+	if members, err := c.etcdClient.MemberList(); err != nil {
+		utilruntime.HandleError(err)
+	} else {
+		bootstrapFound := false
+		for _, member := range members {
+			if member.Name == "etcd-bootstrap" {
+				bootstrapFound = true
+				break
+			}
+		}
+		if !bootstrapFound && len(members) >= 4 {
```

Comment:
Not sure this check is strong enough.

```
2020-05-13T19:46:18.846156291Z E0513 19:46:18.846092       1 host_endpoints_controller.go:126] failed to get etcd client: failed to make etcd client for endpoints [https://10.0.0.6:2379 https://10.0.0.3:2379 https://10.0.0.5:2379]: context deadline exceeded
```

Removing the annotation means the operator loses all awareness of the bootstrap etcd, and in this case we're removing it before the new etcd members have been added to the cluster and quorum achieved.

@hexfusion @retroflexer

Reply:
actually it's not clear from the last CI run what actually removed the annotation, no evidence it was this code

Reply:
found in CI run's etcd operator previous log:

```
2020-05-13T19:45:00.556894239Z I0513 19:45:00.556849       1 event.go:278] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"ccdef0c7-485f-4a19-a2d5-cad88ae3e9c0", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Warning' reason: 'EndpointsUpdated' Updated endpoints/host-etcd-2 -n openshift-etcd because it changed: {"metadata":{"annotations":null},"subsets":[{"addresses":[{"ip":"10.0.0.6","nodeName":"ci-op-j4f2w-m-2.c.openshift-gce-devel-ci.internal"},{"ip":"10.0.0.3","nodeName":"ci-op-j4f2w-m-1.c.openshift-gce-devel-ci.internal"},{"ip":"10.0.0.5","nodeName":"ci-op-j4f2w-m-0.c.openshift-gce-devel-ci.internal"}],"ports":[{"name":"etcd","port":2379,"protocol":"TCP"}]}]}
```

Reply:
Found a fatal flaw in the original patch, latest commit tries another way